### PR TITLE
[linux-port] Correct problem initializations

### PR DESCRIPTION
**Pull request by pow2clk**
Regardless of the order of variables placed in a constructor
member initializer list, the variables are initialized in the
order they are listed in the class or struct. This can cause
issues where the assignment of one variable depends on the other.
For this reason, a warning is produced if the member variables
are listed in a different order. This change moves the member
initializer lists into the order they were being initialized.
Fixes 23 clang and 24 gcc warnings.

Initialization of members of a larger aggregate type may benefit
from braces to clarify which variables are expected to go where.
At the same time, scalars should not be so segregated. This places
or removes such braces wherever appropriate.
Fixes 144 clang warnings

### CODE REVIEW

**Comment by pow2clk** [comment]
Part of an effort to reduce the cavalcade of warnings on Linux build
See https://github.com/google/DirectXShaderCompiler/issues/206

**Comment by ehsannas** [comment]
+1. LGTM.

File: tools/clang/tools/dxcompiler/dxcfilesystem.cpp [hunk 1]
```diff
@@ -307,8 +307,9 @@ class DxcArgsFileSystemImpl : public DxcArgsFileSystem {
 
 public:
   DxcArgsFileSystemImpl(_In_ IDxcBlob *pSource, LPCWSTR pSourceName, _In_opt_ IDxcIncludeHandler* pHandler)
-      : m_pSource(pSource), m_pSourceName(pSourceName), m_includeLoader(pHandler), m_bDisplayIncludeProcess(false),
-        m_pOutputStreamName(nullptr) {
+      : m_pSource(pSource), m_pSourceName(pSourceName), m_pOutputStreamName(nullptr),
+        m_includeLoader(pHandler), m_bDisplayIncludeProcess(false)
+         {
```
Comment (hekota) [inline tools/clang/tools/dxcompiler/dxcfilesystem.cpp]: '{' should be on the previous line

**Comment by pow2clk** [comment]
Done! Thanks!

**Pull request merged**